types/util/validation.d.ts

```diff
@@ -0,0 +1,35 @@
+export declare function eq(
+    valueC: unknown,
+    message?: string
+): (value?: unknown, ctx?: any) => boolean;
+
+export declare function gt(
+    valueC: unknown,
+    message?: string
+): (value?: unknown, ctx?: any) => boolean;
+
+export declare function notEmpty(
+    message?: string
+): (value?: string | unknown[], ctx?: any) => boolean;
+
+export declare function isIn(
+    valueC: unknown[],
+    message?: string
+): (value?: unknown, ctx?: any) => boolean;
```

Comment:
Use generics for a stricter type enforcing.
```suggestion
export declare function eq<T>(
    valueC: T,
    message?: string
): (value?: T, ctx?: any) => boolean;

export declare function gt<T>(
    valueC: T,
    message?: string
): (value?: T, ctx?: any) => boolean;

export declare function notEmpty(
    message?: string
): (value?: string | unknown[], ctx?: any) => boolean;

export declare function isIn<T>(
    valueC: T[],
    message?: string
): (value?: T, ctx?: any) => boolean;
```